ethers-core/src/abi/human_readable.rs

```diff
@@ -463,6 +511,30 @@ fn detect_state_mutability(s: &str) -> StateMutability {
     }
 }
 
+/// Checks if the input `ParamType` contains a `uint8` that the `type_str` also contains `uint8`
+///
+/// Returns `true` if `kind` contains `uint8` but the type_str doesnt
+///
+/// See `AbiParser::parse_type`
+pub(crate) fn is_likely_tuple_not_uint8(kind: &ParamType, type_str: &str) -> bool {
+    if contains_uint8(kind) {
+        !type_str.contains("uint8")
+    } else {
+        false
+    }
+}
+
+/// Returns true if the `ParamType` contains an `uint8`
+pub fn contains_uint8(kind: &ParamType) -> bool {
+    match kind {
+        ParamType::Uint(8) => true,
+        ParamType::Array(arr) => contains_uint8(&*arr),
+        ParamType::FixedArray(arr, _) => contains_uint8(&*arr),
```

Comment:
Think this is better naming? The inner param is not an array itself, it's the inner `ParamType`. It'd be an array if they're nested I guess.

```suggestion
        ParamType::Array(kind) => contains_uint8(&*kind),
        ParamType::FixedArray(kind, _) => contains_uint8(&*kind),
```